### PR TITLE
MTM-53174 - expose multi part POST requests to avoid hardcoded file type

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by c8y-kh4mell — DISMISSED** [review]

**Review by jkot-c8y — DISMISSED** [review]

File: java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java [hunk 1]
```diff
@@ -201,6 +201,15 @@ public <T extends ResourceRepresentation> T putStream(String path, MediaType med
         return parseResponseWithoutId(responseClass, builder.put(stream), OK.getStatusCode());
     }
 
+    @Override
+    public void postStreamWithoutResponse(String path, InputStream inputStream, MediaType inputStreamMediaType) {
+        Builder builder = getResourceBuilder(path);
```
Comment (rohitjoshi-c8y) [inline java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java:206]: **[nitpick]** I would like to be consistent (just like you did for Response object) as I believe in using final liberally :) Feel free to disregard my comment though.

**Review by rohitjoshi-c8y — DISMISSED** [review]

File: java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java (hunk 1 above)
Comment (ghost) [reply java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java:206]: final removed completely to be consistent with other methods :)

Comment (rohitjoshi-c8y) [reply java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java:206]: It's the other way I wanted to go but that's fine :D

**Review by rohitjoshi-c8y — APPROVED** [review]

File: java-client/src/main/java/com/cumulocity/sdk/client/RestOperations.java [hunk 2]
```diff
@@ -30,6 +30,8 @@ <T extends ResourceRepresentation> T putStream(String path, String contentType,
     <T extends ResourceRepresentation> T putStream(String path, MediaType mediaType, InputStream content,
                                                    Class<T> responseClass);
 
+    void postStreamWithoutResponse(String path, InputStream inputStream, MediaType inputStreamMediaType);
```
Comment (karol-c8y) [inline java-client/src/main/java/com/cumulocity/sdk/client/RestOperations.java:33]: I'm not huge fan of such names, postStream() should be fine, "withoutRresponse" is visible from method signature :)

**Review by karol-c8y — APPROVED** [review]

File: java-client/src/main/java/com/cumulocity/sdk/client/RestOperations.java (hunk 2 above)
Comment (ghost) [reply java-client/src/main/java/com/cumulocity/sdk/client/RestOperations.java:33]: yee I would also didn't thought about such naming but I just blindly follow the convention which was already in the file. Maybe it doesn't make sense to multiply bad practice.
Here you go sir - #350

**Pull request merged**